lib/hotel_manager.rb

```diff
@@ -0,0 +1,90 @@
+require 'date'
+require_relative 'room'
+require_relative 'reservation'
+require_relative 'date_range'
+require_relative 'block'
+
+module HotelManagementSystem
+    class HotelManager
+        attr_reader :reservations
+
+        COST = 200
+
+        def initialize
+            @reservations = []
+            @rooms = []
+            @blocks = []
+
+            1..20.times do |i|
+                room = HotelManagementSystem::Room.new(room: i + 1)
+                @rooms << room
+            end
+        end
+
+        def list_rooms
+            return @rooms
+        end
+
+        def reserve(start_date, end_date)
+            if !HotelManagementSystem::DateRange.is_valid?(start_date, end_date)
+                raise ArgumentError, "You have entered an invalid date range."
+            end
+            
+            # assigns first available room to reservation
+            room_assignment = list_available_rooms(start_date, end_date).first
+            raise ArgumentError, "There is no room available for these dates." if room_assignment == nil
```

Comment:
Very slightly cleaner.

```suggestion
            raise ArgumentError, "There is no room available for these dates." unless room_assignment
```